docs/testing.md

```diff
@@ -29,6 +29,35 @@ agent1 ->  agent/worker node
 
 Note/TODO: k3s external db fails working with etcd only node. Refer: https://docs.k3s.io/datastore/ha
 
+### Secret-Encryption tests Setup Requirements/Assumptions
+We need a split role setup for this test:
```

Comment:
Do we want to restrict the tests to have this config? We should be able to run the test in non-split mode as well, right?

Reply:
fixed. Added more verbage to suggest we can test on both. Made it clear, that the split role is for patch validation test case.